src/samples/doxygen/buffer.cpp

```diff
@@ -0,0 +1,52 @@
+#include <rpp/rpp.hpp>
+#include <iostream>
+
+std::ostream & operator<<(std::ostream & out, const std::vector<int>& list)
+{
+    out << "{";
+    auto size = list.size();
+    for (int i = 0; i < size; ++i)
+    {
+        out << list.at(i);
+        if (i < size - 1)
+        {
+            out << ",";
+        }
+    }
+    out << "}";
+
+    return out;
+};
+
+/**
+ * \example buffer.cpp
+ **/
+int main()
+{
+    //! [buffer]
+    {
+        // The stream that uses rvalue overloads for operators
+        rpp::source::just(1, 2, 3, 4, 5)
+            .buffer(2)
+            .subscribe(
+                [](const std::vector<int> &v) { std::cout << v << "-"; },
+                [](const std::exception_ptr &error) {},
+                []() { std::cout << "|" << std::endl; });
+        // Source: -1-2-3-4-5--|
+        // Output: {1,2}-{3,4}-|
+    }
+
+    {
+        // The stream that uses lvalue overloads for operators
+        auto source = rpp::source::just(1, 2, 3, 4, 5);
+        source.buffer(2)
+            .subscribe(
+                [](const std::vector<int> &v) { std::cout << v << "-"; },
+                [](const std::exception_ptr &error) {},
+                []() { std::cout << "|" << std::endl; });
+        // Source: -1-2-3-4-5--|
+        // Output: {1,2}-{3,4}-|
+    }
```

Comment:
Any difference?

Reply:
The only difference is the 1st chain uses the rvalue overload and 2nd chain uses the lvalue. I just want to point out they are different overloads and people could break one while updating the other.

Reply:
Oh, i see. But then it is better to place it inside tests, not doxygen sample. Because samples used as API reference and usage example for users. For me as for user it would be not clear what is difference in this case

Reply:
Good point!